CMakeLists.txt

```diff
@@ -171,10 +171,6 @@ if (WITH_CUDA)
         list(APPEND CUDA_NVCC_FLAGS "-D_MWAITXINTRIN_H_INCLUDED -D_FORCE_INLINES")
       endif()
       SET(CUDA_PROPAGATE_HOST_FLAGS OFF)
-
-      if (APPLE)
-        set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -Xcompiler -isysroot -Xcompiler ${CMAKE_OSX_SYSROOT}")
-      endif()
```

Comment:
Don’t remove this. The sysroot is needed in certain cases, e.g. with Xcode 9 on macOS 10.12.

Reply:
okay, but it breaks the build for the current clang version. How to make this future save?

Reply:
I wasn't able to reproduce your issue on a real Mac with the current python branch, but I came up with a simple fix that checks whether `CMAKE_OSX_SYSROOT` is even set. Just merge https://github.com/KaiSzuttor/espresso/pull/17